nifi-nar-bundles/nifi-media-bundle/nifi-media-processors/src/main/java/org/apache/nifi/processors/media/ExtractMediaMetadata.java

```diff
@@ -0,0 +1,276 @@
+            .name("File Name Filter")
+            .description("A regular expression identifying file names which metadata should extracted.  As flowfiles"
+                    + " are processed, if the file name matches this regular expression or this expression is"
+                    + " blank, the flowfile will be scanned for it's MIME type and metadata.  If left blank, all"
+                    + " flowfiles will be scanned.")
+            .required(false)
+            .addValidator(StandardValidators.REGULAR_EXPRESSION_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor MIME_TYPE_FILTER = new PropertyDescriptor.Builder()
+            .name("MIME Type Filter")
+            .description("A regular expression identifying MIME types for which metadata should extracted.  Flowfiles"
+                    + " selected for scanning by the File Name Filter are parsed to determine the MIME type and extract"
+                    + " metadata.  If the MIME type found matches this regular expression or this expression is"
+                    + " blank, the metadata keys that match the Metadata Key Filter will be added to the flowfile"
+                    + " attributes.  There is no guarantee in what order attributes will be produced.  If"
+                    + " left blank, metadata will be extracted from all flow files selected for scanning.")
+            .required(false)
+            .addValidator(StandardValidators.REGULAR_EXPRESSION_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor METADATA_KEY_FILTER = new PropertyDescriptor.Builder()
+            .name("Metadata Key Filter")
+            .description("A regular expression identifying which metadata keys received from the parser should be"
+                    + " added to the flowfile attributes.  If left blank, all metadata keys parsed will be added to the"
+                    + " flowfile attributes.")
+            .required(false)
+            .addValidator(StandardValidators.REGULAR_EXPRESSION_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor METADATA_KEY_PREFIX = new PropertyDescriptor.Builder()
+            .name("Metadata Key Prefix")
+            .description("Text to be prefixed to metadata keys as the are added to the flowfile attributes.  It is"
+                    + " recommended to end with with a separator character like '.' or '-', this is not automatically "
+                    + " added by the processor.")
+            .required(false)
+            .addValidator(StandardValidators.ATTRIBUTE_KEY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    public static final Relationship SUCCESS = new Relationship.Builder()
+            .name("success")
+            .description("Any FlowFile that successfully has image metadata extracted will be routed to success")
```

Comment:
The original version I created was just for WAV files, "image" needs to be removed now.